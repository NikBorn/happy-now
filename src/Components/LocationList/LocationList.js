import React from 'react';
import LocationCard from '../LocationCard/LocationCard.js';


const  LocationList = (props) => {
  console.log('count: ', props.count)

  const listObjs = props.locations.map(locationInfo => {
    return <LocationCard 
      key={locationInfo.id} 
      locationInfo = { locationInfo } />;
  }); 

  const displayLocationCards = listObjs.filter((location, index) => {
    if (index < props.count) {
      return location;
    } else {
      return null;
    }
  })
  
  return (
    <div className='location-list'>
      {displayLocationCards }
    </div>
  );
};



export default LocationList;