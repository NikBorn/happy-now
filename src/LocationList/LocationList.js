import React from 'react';
import LocationCard from '../LocationCard/LocationCard.js';
import { connect } from 'react-redux';


const  LocationList = (props) => {
  const listObjs = props.locations.map(locationInfo => {
    return <LocationCard 
      key={locationInfo.id} 
      locationInfo = { locationInfo } />;
  }); 

  return (
    <div>
      { listObjs }
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    locations: state.locations
  };
};

export default connect(mapStateToProps, null)(LocationList);