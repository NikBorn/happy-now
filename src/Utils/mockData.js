

export const mockLocationsResponse = [
  {
    isFavorite: false,
    id: "4b044f71f964a5202e5322e3",
    name: "The Glenn",
    contact: {
      phone: "3032551400",
      formattedPhone: "(303) 255-1400",
      twitter: "biggyglenn",
      facebook: "115875251829275",
      facebookName: "Glenn Bar"},
    location: {
      address: "11140 Irma Dr",
      lat: 39.89855629220959,
      lng: -104.96589290871326,
      labeledLatLngs: [{
        label: "display",
        lat: 39.89855629220959,
        lng: -104.96589290871326
      }],
      distance: 4876,
      postalCode: "80233",
      cc: "US",
      city: "Northglenn",
      state: "CO",
      country: "United States",
      formattedAddress: ['11140 Irma Dr', 'Northglenn, CO 80233', 'United States']},
    categories: [{
      id: "4bf58dd8d48988d116941735",
      name: "Bar",
      pluralName: "Bars",
      shortName: "Bar",
      icon: {
        prefix: "https://ss3.4sqi.net/img/categories_v2/nightlife/pub_",
        suffix: ".png"
      },
      primary: true}],
    verified: false,
    stats: {
      checkinsCount: 2080,
      usersCount: 702,
      tipCount: 29},
    url: "http://theglennbarandgrill.com",
    price: {
      tier: 2,
      message: "Moderate",
      currency: "$"},
    hasMenu: true,
    rating: 8.4,
    ratingSignals: 84,
    menu: { 
      type: "Menu", 
      label: "Menu", 
      anchor: "View Menu",
      url: "https://foursquare.com/v/the-glenn/4b044f71f964a5202e5322e3/menu",
      mobileUrl: "https://foursquare.com/v/4b044f71f964a5202e5322e3/device_menu"},
    allowMenuUrlEdit: true,
    beenHere: {
      count: 0,
      marked: false,
      lastCheckinExpiredAt: 0},
    hours: {
      status: "Likely open",
      richStatus: {
        entities: [],
        text: "Likely open"},
      isOpen: true,
      isLocalHoliday: false},
    photos: {
      count: 0,
      groups: []},
    hereNow: {
      count: 0,
      summary: "Nobody here",
      groups: [] }},
  {
    name: "Tequila Rock",
    price: { tier: 2, message: "Moderate", currency: "$" },
    rating: 6.4,
    ratingSignals: 399,
    contact: { phone: "3032551400", 
      formattedPhone: "(303) 255-1400", 
      twitter: "biggyglenn", 
      facebook: "115875251829275", 
      facebookName: "Glenn Bar" },
    isFavorite: false,
    id: "4d8a35c976e1236a99721534"
  },
  {
    name: "Paddy's Pub",
    price: { tier: 1, message: "Moderate", currency: "$" },
    rating: 3.4,
    ratingSignals: 107,
    contact: { phone: "3038675309", 
      formattedPhone: "(303) 867-5309", 
      twitter: "biggyglenn", 
      facebook: "115875251829275", 
      facebookName: "Glenn Bar" },
    isFavorite: false,
    id: "4d8a35c976e1236a99725553"
  }
];

