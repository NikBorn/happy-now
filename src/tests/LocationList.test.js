import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import LocationList from '../Components/LocationList/LocationList.js';
import Adapter from 'enzyme-adapter-react-16';
import { MockLocation, mockLocationsResponse } from '../Utils/mockData.js';
import configureStore from 'redux-mock-store';

configure({ adapter: new Adapter() });

describe('LocationList', () => {
  const mockStore = configureStore();
  const stateLocations = mockLocationsResponse;
  const initialState = {
    locations: stateLocations,
    userLocation: {},
    activeUser: null,
    count: 10,
    favorites: []
  };
  const store = mockStore(initialState);
  const wrapper = shallow(
    <LocationList store={store} locations={stateLocations} />);

  it('shold always match the Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});