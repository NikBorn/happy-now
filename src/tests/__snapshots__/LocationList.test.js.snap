// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationList shold always match the Snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LocationList
    locations={
        Array [
            Object {
              "allowMenuUrlEdit": true,
              "beenHere": Object {
                "count": 0,
                "lastCheckinExpiredAt": 0,
                "marked": false,
              },
              "categories": Array [
                Object {
                  "icon": Object {
                    "prefix": "https://ss3.4sqi.net/img/categories_v2/nightlife/pub_",
                    "suffix": ".png",
                  },
                  "id": "4bf58dd8d48988d116941735",
                  "name": "Bar",
                  "pluralName": "Bars",
                  "primary": true,
                  "shortName": "Bar",
                },
              ],
              "contact": Object {
                "facebook": "115875251829275",
                "facebookName": "Glenn Bar",
                "formattedPhone": "(303) 255-1400",
                "phone": "3032551400",
                "twitter": "biggyglenn",
              },
              "hasMenu": true,
              "hereNow": Object {
                "count": 0,
                "groups": Array [],
                "summary": "Nobody here",
              },
              "hours": Object {
                "isLocalHoliday": false,
                "isOpen": true,
                "richStatus": Object {
                  "entities": Array [],
                  "text": "Likely open",
                },
                "status": "Likely open",
              },
              "id": "4b044f71f964a5202e5322e3",
              "isFavorite": false,
              "location": Object {
                "address": "11140 Irma Dr",
                "cc": "US",
                "city": "Northglenn",
                "country": "United States",
                "distance": 4876,
                "formattedAddress": Array [
                  "11140 Irma Dr",
                  "Northglenn, CO 80233",
                  "United States",
                ],
                "labeledLatLngs": Array [
                  Object {
                    "label": "display",
                    "lat": 39.89855629220959,
                    "lng": -104.96589290871326,
                  },
                ],
                "lat": 39.89855629220959,
                "lng": -104.96589290871326,
                "postalCode": "80233",
                "state": "CO",
              },
              "menu": Object {
                "anchor": "View Menu",
                "label": "Menu",
                "mobileUrl": "https://foursquare.com/v/4b044f71f964a5202e5322e3/device_menu",
                "type": "Menu",
                "url": "https://foursquare.com/v/the-glenn/4b044f71f964a5202e5322e3/menu",
              },
              "name": "The Glenn",
              "photos": Object {
                "count": 0,
                "groups": Array [],
              },
              "price": Object {
                "currency": "$",
                "message": "Moderate",
                "tier": 2,
              },
              "rating": 8.4,
              "ratingSignals": 84,
              "stats": Object {
                "checkinsCount": 2080,
                "tipCount": 29,
                "usersCount": 702,
              },
              "url": "http://theglennbarandgrill.com",
              "verified": false,
            },
            Object {
              "contact": Object {
                "facebook": "115875251829275",
                "facebookName": "Glenn Bar",
                "formattedPhone": "(303) 255-1400",
                "phone": "3032551400",
                "twitter": "biggyglenn",
              },
              "id": "4d8a35c976e1236a99721534",
              "isFavorite": false,
              "name": "Tequila Rock",
              "price": Object {
                "currency": "$",
                "message": "Moderate",
                "tier": 2,
              },
              "rating": 6.4,
              "ratingSignals": 399,
            },
            Object {
              "contact": Object {
                "facebook": "115875251829275",
                "facebookName": "Glenn Bar",
                "formattedPhone": "(303) 867-5309",
                "phone": "3038675309",
                "twitter": "biggyglenn",
              },
              "id": "4d8a35c976e1236a99725553",
              "isFavorite": false,
              "name": "Paddy's Pub",
              "price": Object {
                "currency": "$",
                "message": "Moderate",
                "tier": 1,
              },
              "rating": 3.4,
              "ratingSignals": 107,
            },
          ]
    }
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
      "className": "location-list",
    },
    "ref": null,
    "rendered": Array [],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
        "className": "location-list",
      },
      "ref": null,
      "rendered": Array [],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
